rfcs/0138-rfc-repos.md

```diff
@@ -0,0 +1,106 @@
+>
+> To review the RFC
+> - Open https://github.com/$owner/$repo/blob/master/README.md?plain=1
+> - Select the lines you wish to comment on
+> - Click the "..." menu on the left
+> - Select "Reference in new issue" to open a new issue commenting on those lines
+>
+> To suggest changes, open a PR against the repository.
+>
+> All issues and pull requests must be resolved before the FCP can be initiated and completed.
+>
+> This issue may only be used for RFC meta-discussions, such as shepherd nominations, FCP periods, meeting schedules, etc.
+
+The FCP must be initiated on a specific commit of the repository.
+When the FCP passes, the repositories contents are committed to the [rfcs](https://github.com/NixOS/rfcs) repository by the [RFC steering committee](https://github.com/NixOS/rfcs#rfc-steering-committee) with a commit that closes the original issue.
+
+## Workflow to get RFC updates
+To get updates for RFC's, instead of subscribing to the PR, one has to watch the repository.
+
+## Repository transfer
+
+Once the RFC is merged, the repository has to be transferred to the NixOS organization under https://github.com/NixOS/rfc-NUMBER. This is to ensure the discussions aren't lost in the future. The repository will then be archived.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+- There's no line-comment view anymore (the "Files changed" tab for PR's), where you can see the entire proposal and all (up-to-date) line comments.
+- Commenting on an RFC won't automatically subscribe you to updates anymore.
+- Less separation between RFC authors and reviewers with edit suggestions, making it harder to decide who may be a shepherd, since authors can't be shepherds.
+- Force pushes won't be visible anymore in the history.
+  - Force pushing shouldn't be necessary for RFCs, since they always get squash merged anyways
+    - With the previous approach, perhaps the main use case of force pushing was to reduce the PR items to be loaded,
+      since commits also count towards the number of items loaded by GitHub.
+      This is obviously not necessary anymore.
+  - It would be good to protect the main branch for active RFCs,
+    which could be done by requiring the RFC repo be moved to the NixOS organisation.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Optional instead of required
+
+Instead of requiring this process, it could be opt-in for "bigger" RFCs.
+- (-) It's not clear how to decide whether an RFC should have a repository, there's no way to know how big discussions become in advance.
+
+## Fork branch instead of separate repository
+
+Instead of creating a new repository for each RFC, a new branch in a fork can be created instead.
+- (-) Means that a single GitHub user/organization can't have more than one RFC open at a time without mixing of issues/PRs occurring (since GitHub only supports having a single fork of a repository).
```

Comment:
Is this a big issue? Have there been instances in the past where the same person is authoring multiple RFCs simultaneously? And if so, would labels on both issues & PRs by RFC number not solve this problem satisfactorily?

Reply:
Unclear. Yes, e.g. 45/46. Maybe.

To be fair, line comments get auto-obsoleted on updates but issues are not; so if a full rewrite test-wise (which 46 has seen) would need probably need labelling even within a single RFC.

Reply:
I'm kind of already using this approach and I already have 4 RFC repos :)
- This one
- #166 (https://github.com/nix-rfc-101/rfcs) as co-author
- https://github.com/nix-rfc-canonical-domain/rfcs
- https://github.com/tweag/epcb

I think it would be much cleaner to separate all RFCs with a repository. There's various benefits, like being able to infer that an RFC is likely ready for FCP once issues and PRs are at 0. Or not having to worry about having the correct label. Or the workflow not changing when somebody opens another RFC (before you wouldn't have needed a filter, but now you do), etc.

Reply:
Yeah, that makes sense. Might not be worth exploring this alternative, in that case.